src/mlpack/core/dists/gamma_distribution.cpp

```diff
@@ -117,3 +135,84 @@ void GammaDistribution::Train(const arma::vec& logMeanxVec,
     beta(row) = meanx / aEst;
   }
 }
+
+// Returns the probability of the provided observations.
+void GammaDistribution::Probability(const arma::mat& observations, 
+                                    arma::vec& probabilities) const
+{
+  size_t numObs = observations.n_cols;
+
+  // Set all equal to 1 (multiplication neutral).
+  probabilities.ones(numObs);
+
+  // Compute denominator only once for each dimension.
+  arma::vec denominators(alpha.n_elem);
+  for (size_t d = 0; d < alpha.n_elem; ++d)
+    denominators(d) = std::tgamma(alpha(d)) * std::pow(beta(d), alpha(d));
+
+  // Compute probability of each observation.
+  for (size_t i = 0; i < numObs; ++i)
+  {
+    for (size_t d = 0; d < observations.n_rows; ++d)
+    {
+      // Compute probability using Multiplication Law.
+      probabilities(i) *= 
+        std::pow(observations(d, i), alpha(d) - 1) *
+        std::exp(-observations(d, i) / beta(d)) /
+        denominators(d);
+    }
+  }
+}
+
+// Returns the probability of one observation (x) for one of the Gamma's
+// dimensions.
+double GammaDistribution::Probability(double x, size_t dim) const
+{
+  return 
+    std::pow(x, alpha(dim) - 1) * std::exp(-x / beta(dim)) / 
+    (std::tgamma(alpha(dim)) * std::pow(beta(dim), alpha(dim)));
```

Comment:
I know I'm being picky but I think the indentation needs to be one more here.  Another option is to split this into multiple lines for readability, but that's up to you.  Realistically, to the compiler, it shouldn't make a difference if you compute a bunch of doubles then multiply them together, or compute the result all at once.